src/storm/mesos/MesosNimbus.java

```diff
@@ -547,29 +547,20 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
 
               String executorDataStr = JSONValue.toJSONString(executorData);
               LOG.info("Launching task with Mesos Executor data: <" + executorDataStr + ">");
+
+              // Create a docker builder if a docker image was defined in the yaml
+              Optional<ContainerInfo.Builder> containerInfoBuilder = createDockerInfoBuilder(getDockerImage());
```

Comment:
Why have this in `assignSlots()`?   i.e., if you just call it from within `createExecutorInfo()` then you can avoid this extra parameter, and also slightly shrink the crazy-length of the `assignSlots()` method.